### ISSUE TITLE
[BUG] - InvalidBlock on testnet with 1.35.3

### ISSUE THREAD

**Issue by andreabravetti** [input-output-hk/cardano-node#4252]
**Summary**
I just updated a testnet node to 1.35.3 (not a stake pool, just a full node) and I see these errors:

```
ago 02 09:39:21 frodo cardano-node[2072]: [frodo:cardano.node.ChainDB:Notice:58] [2022-08-02 07:39:21.94 UTC] Chain extended, new tip: 89561bc578e49e5678548bfaec3b77cfcef508f735b8ef85b529abc0e5a46c67 at slot 64397182
ago 02 09:39:21 frodo cardano-node[2072]: [frodo:cardano.node.ChainDB:Notice:58] [2022-08-02 07:39:21.95 UTC] Chain extended, new tip: 722bbd1b015dec7de8f1a21288d45f4a3ec0393017e1006a1f0a29c8b710f4e6 at slot 64397191
ago 02 09:39:21 frodo cardano-node[2072]: [frodo:cardano.node.ChainDB:Notice:58] [2022-08-02 07:39:21.95 UTC] Chain extended, new tip: da5d3e4f7365d4793d981483bb7f4146aa8259d33e2e2ae09258574a437b703a at slot 64397202
ago 02 09:39:22 frodo cardano-node[2072]: [frodo:cardano.node.ChainDB:Error:58] [2022-08-02 07:39:22.14 UTC] Invalid block 470b412e2ab520b348f4ddd200bf3b8cf51fda49cdf449cba07a38f0d27c7a6b at slot 64400487: ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "f79a9c7749dccb4cba4f4e2b1d395503abf3b21afa39887c3fe60254"])))))])})))))))
ago 02 09:39:22 frodo cardano-node[2072]: [frodo:cardano.node.ChainDB:Info:58] [2022-08-02 07:39:22.14 UTC] Valid candidate a39c50470f6615ee765d65e532a44f48f26c921932bcabd13ee4e6ac09826532 at slot 64400358
ago 02 09:39:22 frodo cardano-node[2072]: [frodo:cardano.node.ChainDB:Notice:58] [2022-08-02 07:39:22.19 UTC] Chain extended, new tip: a39c50470f6615ee765d65e532a44f48f26c921932bcabd13ee4e6ac09826532 at slot 64400358
```

It seems blocked here, no progress.

Then I restarted the node and now it repeat this every few seconds:

```
ago 02 09:46:41 frodo cardano-node[2113]: [frodo:cardano.node.DnsSubscription:Notice:817] [2022-08-02 07:46:41.46 UTC] Domain: "relays-new.cardano-testnet.iohkdev.io" Connection Attempt End, destination 184.169.196.177:3001 outcome: ConnectSuccess
ago 02 09:46:41 frodo cardano-node[2113]: [frodo:cardano.node.DnsSubscription:Notice:818] [2022-08-02 07:46:41.55 UTC] Domain: "relays-new.cardano-testnet.iohkdev.io" Connection Attempt End, destination 54.199.108.71:3001 outcome: ConnectSuccessLast
ago 02 09:46:42 frodo cardano-node[2113]: [frodo:cardano.node.DnsSubscription:Error:817] [2022-08-02 07:46:42.26 UTC] Domain: "relays-new.cardano-testnet.iohkdev.io" Application Exception: 184.169.196.177:3001 InvalidBlock (At (Block {blockPointSlot = SlotNo 64400532, blockPointHash = 87d7927b3e8096d9ba6a5c189f6c221aab5d42032fba49c6c0d50d1aec8f1e6c})) 470b412e2ab520b348f4ddd200bf3b8cf51fda49cdf449cba07a38f0d27c7a6b (ValidationError (ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "f79a9c7749dccb4cba4f4e2b1d395503abf3b21afa39887c3fe60254"])))))])})))))))))
ago 02 09:46:42 frodo cardano-node[2113]: [frodo:cardano.node.ErrorPolicy:Warning:78] [2022-08-02 07:46:42.26 UTC] IP 184.169.196.177:3001 ErrorPolicySuspendPeer (Just (ApplicationExceptionTrace (InvalidBlock (At (Block {blockPointSlot = SlotNo 64400532, blockPointHash = 87d7927b3e8096d9ba6a5c189f6c221aab5d42032fba49c6c0d50d1aec8f1e6c})) 470b412e2ab520b348f4ddd200bf3b8cf51fda49cdf449cba07a38f0d27c7a6b (ValidationError (ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "f79a9c7749dccb4cba4f4e2b1d395503abf3b21afa39887c3fe60254"])))))])})))))))))))) 200s 200s
ago 02 09:46:42 frodo cardano-node[2113]: [frodo:cardano.node.DnsSubscription:Error:818] [2022-08-02 07:46:42.88 UTC] Domain: "relays-new.cardano-testnet.iohkdev.io" Application Exception: 54.199.108.71:3001 InvalidBlock (At (Block {blockPointSlot = SlotNo 64400532, blockPointHash = 87d7927b3e8096d9ba6a5c189f6c221aab5d42032fba49c6c0d50d1aec8f1e6c})) 470b412e2ab520b348f4ddd200bf3b8cf51fda49cdf449cba07a38f0d27c7a6b (ValidationError (ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "f79a9c7749dccb4cba4f4e2b1d395503abf3b21afa39887c3fe60254"])))))])})))))))))
ago 02 09:46:42 frodo cardano-node[2113]: [frodo:cardano.node.ErrorPolicy:Warning:78] [2022-08-02 07:46:42.88 UTC] IP 54.199.108.71:3001 ErrorPolicySuspendPeer (Just (ApplicationExceptionTrace (InvalidBlock (At (Block {blockPointSlot = SlotNo 64400532, blockPointHash = 87d7927b3e8096d9ba6a5c189f6c221aab5d42032fba49c6c0d50d1aec8f1e6c})) 470b412e2ab520b348f4ddd200bf3b8cf51fda49cdf449cba07a38f0d27c7a6b (ValidationError (ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "f79a9c7749dccb4cba4f4e2b1d395503abf3b21afa39887c3fe60254"])))))])})))))))))))) 200s 200s
```

**Steps to reproduce**
I produced this error once with a testnet node synchronized to something slightly earlier than slot 64400358, updated to 1.35.3 from hydra builds (https://hydra.iohk.io/build/17329548) and restarted.

**Expected behavior**
It should sync without errors.

**System info (please complete the following information):**
- OS Name: Ubuntu 
- OS Version: 20.04.4 LTS
- Node version: cardano-node 1.35.3 - linux-x86_64 - ghc-8.10 git rev b443f540133888442e5475c05890dc2eee13d43e
- CLI version: cardano-cli 1.35.3 - linux-x86_64 - ghc-8.10 git rev b443f540133888442e5475c05890dc2eee13d43e

**Comment by SebastienGllmt** [comment]
Can confirm I get the same issue

**Comment by SebastienGllmt** [comment]
I tried dropping recently blocks (by deleted recent files in the `immutable` folder) and resyncing and it failed again with the same issue

**Comment by andreabravetti** [comment]
I'm trying to sync from start with both 1.35.2 and 1.35.3, I'll report ASAP any finding.

**Comment by andreabravetti** [comment]
I can add that 1.35.2 sync at 100% without problems.

**Comment by andreabravetti** [comment]
Restarting 1.35.3 from the beginning is halt on the same block.

Updating to 1.35.3 on an already updated db, after slot 64983250, it work and keep syncing, so you may not notice the problem if you update on an already in sync node.

So, as far as I understand, to replicate you must start from something before slot 64397202.

**Comment by jfruiz11375** [comment]
I am seeing the same issue but I was testing the update from a 1.35.1 node that was out of sync. restarted but same issue. I ut this in the 1.35.3 SCOPE but moved this here now that i found this bug already reported

[cardanon:cardano.node.ErrorPolicy:Notice:465] [2022-08-03 22:39:15.89 UTC] IP 3.72.26.163:3001 ErrorPolicySuspendConsumer (Just (ConnectionExceptionTrace (SubscriberError {seType = SubscriberParallelConnectionCancelled, seMessage = "Parallel connection cancelled", seStack = []}))) 1s
[cardanon:cardano.node.ErrorPolicy:Notice:465] [2022-08-03 22:39:15.89 UTC] IP 18.198.216.177:3001 ErrorPolicySuspendConsumer (Just (ConnectionExceptionTrace (SubscriberError {seType = SubscriberParallelConnectionCancelled, seMessage = "Parallel connection cancelled", seStack = []}))) 1s
[cardanon:cardano.node.ErrorPolicy:Notice:465] [2022-08-03 22:39:15.89 UTC] IP 3.72.137.52:3001 ErrorPolicySuspendConsumer (Just (ConnectionExceptionTrace (SubscriberError {seType = SubscriberParallelConnectionCancelled, seMessage = "Parallel connection cancelled", seStack = []}))) 1s
[cardanon:cardano.node.ErrorPolicy:Notice:465] [2022-08-03 22:39:15.89 UTC] IP 3.131.32.242:3001 ErrorPolicySuspendConsumer (Just (ConnectionExceptionTrace (SubscriberError {seType = SubscriberParallelConnectionCancelled, seMessage = "Parallel connection cancelled", seStack = []}))) 1s
[cardanon:cardano.node.ErrorPolicy:Notice:465] [2022-08-03 22:39:15.89 UTC] IP 54.199.108.71:3001 ErrorPolicySuspendConsumer (Just (ConnectionExceptionTrace (SubscriberError {seType = SubscriberParallelConnectionCancelled, seMessage = "Parallel connection cancelled", seStack = []}))) 1s
[cardanon:cardano.node.ChainDB:Notice:444] [2022-08-03 22:39:16.14 UTC] Chain extended, new tip: f58545f02a8961b17256a888ea2722bcf60fd4fe25e317daa96c6391048d97f2 at slot 64387391
[cardanon:cardano.node.ChainDB:Notice:444] [2022-08-03 22:39:18.23 UTC] Chain extended, new tip: f11d161b26d73e2ef6b7d96e53cabba8a2d3305c3fa162e0ab0a6642188d5393 at slot 64387992
[cardanon:cardano.node.ChainDB:Notice:444] [2022-08-03 22:39:19.48 UTC] Chain extended, new tip: 70e41bf110741eafef0dd821af33eab28ec04d811bfc7113e499ad32437e5a1c at slot 64394492
[cardanon:cardano.node.ChainDB:Notice:444] [2022-08-03 22:39:20.76 UTC] Chain extended, new tip: 9295bc64291688bb0c9b4f0af92d7fb37f1ae670956b71cbaa44904bb61d1834 at slot 64400330
[cardanon:cardano.node.ChainDB:Error:444] [2022-08-03 22:39:20.78 UTC] Invalid block 470b412e2ab520b348f4ddd200bf3b8cf51fda49cdf449cba07a38f0d27c7a6b at slot 64400487: ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "f79a9c7749dccb4cba4f4e2b1d395503abf3b21afa39887c3fe60254"])))))])})))))))
[cardanon:cardano.node.ChainDB:Info:444] [2022-08-03 22:39:20.78 UTC] Valid candidate a39c50470f6615ee765d65e532a44f48f26c921932bcabd13ee4e6ac09826532 at slot 64400358

**Comment by jfruiz11375** [comment]
Now just keeps repeating 

[cardanon:cardano.node.DnsSubscription:Error:788] [2022-08-04 01:06:31.37 UTC] Domain: "relays-new.cardano-testnet.iohkdev.io" Application Exception: 184.169.196.177:3001 InvalidBlock (At (Block {blockPointSlot = SlotNo 64400532, blockPointHash = 87d7927b3e8096d9ba6a5c189f6c221aab5d42032fba49c6c0d50d1aec8f1e6c})) 470b412e2ab520b348f4ddd200bf3b8cf51fda49cdf449cba07a38f0d27c7a6b (ValidationError (ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "f79a9c7749dccb4cba4f4e2b1d395503abf3b21afa39887c3fe60254"])))))])})))))))))
[cardanon:cardano.node.ErrorPolicy:Warning:471] [2022-08-04 01:06:31.37 UTC] IP 184.169.196.177:3001 ErrorPolicySuspendPeer (Just (ApplicationExceptionTrace (InvalidBlock (At (Block {blockPointSlot = SlotNo 64400532, blockPointHash = 87d7927b3e8096d9ba6a5c189f6c221aab5d42032fba49c6c0d50d1aec8f1e6c})) 470b412e2ab520b348f4ddd200bf3b8cf51fda49cdf449cba07a38f0d27c7a6b (ValidationError (ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "f79a9c7749dccb4cba4f4e2b1d395503abf3b21afa39887c3fe60254"])))))])})))))))))))) 200s 200s
[cardanon:cardano.node.DnsSubscription:Error:789] [2022-08-04 01:06:31.42 UTC] Domain: "relays-new.cardano-testnet.iohkdev.io" Application Exception: 3.131.32.242:3001 InvalidBlock (At (Block {blockPointSlot = SlotNo 64400532, blockPointHash = 87d7927b3e8096d9ba6a5c189f6c221aab5d42032fba49c6c0d50d1aec8f1e6c})) 470b412e2ab520b348f4ddd200bf3b8cf51fda49cdf449cba07a38f0d27c7a6b (ValidationError (ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "f79a9c7749dccb4cba4f4e2b1d395503abf3b21afa39887c3fe60254"])))))])})))))))))
[cardanon:cardano.node.ErrorPolicy:Warning:471] [2022-08-04 01:06:31.42 UTC] IP 3.131.32.242:3001 ErrorPolicySuspendPeer (Just (ApplicationExceptionTrace (InvalidBlock (At (Block {blockPointSlot = SlotNo 64400532, blockPointHash = 87d7927b3e8096d9ba6a5c189f6c221aab5d42032fba49c6c0d50d1aec8f1e6c})) 470b412e2ab520b348f4ddd200bf3b8cf51fda49cdf449cba07a38f0d27c7a6b (ValidationError (ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "f79a9c7749dccb4cba4f4e2b1d395503abf3b21afa39887c3fe60254"])))))])})))))))))))) 200s 200s

**Comment by btbf** [comment]
@jfruiz11375 
This will be fixed in 1.35.3-rc1.
The current recommended version of the public testnet is 1.35.2.

Please see the following announcement for details
https://t.me/SPOannouncements/176

**Comment by trevinwisaksana** [comment]
Hi, I'm still having this issue in 1.35.3-rc1. To replicate this you have to try to sync the node from scratch. I'm using an M1 MacBook Pro. This is where I got the invalid block:
`
[Trevins-:cardano.node.ChainDB:Error:34] [2022-08-07 23:23:34.40 UTC] Invalid block 708745ea94515b679223a5514e6e6dba068208e5b79780134206f74e1e79e727 at slot 63419637: ExtValidationErrorLedger (HardForkLedgerErrorFromEra S (S (S (S (S (Z (WrapLedgerErr {unwrapLedgerErr = BBodyError (BlockTransitionError [ShelleyInAlonzoPredFail (LedgersFailure (LedgerFailure (UtxowFailure (MalformedScriptWitnesses (fromList [ScriptHash "eee7bfb25cfbbbcd31a1d5d648429ff919430ee5d06d8d056e48d605"])))))])})))))))
`

**Comment by jfruiz11375** [comment]
I don't believe the 1.35.3-rc1 is open for the public testnet yet.

**Comment by trevinwisaksana** [comment]
I see, I'll revert back to 1.35.2. I assumed it's open for public testnet because they already have 1.35.3-testnetonly. Thanks

**Comment by andreabravetti** [comment]
I think that this issue can be closed here: What happened is well known, the issue was fixed with new testnets and 1.35.3 is fine. No need to keep this open.
